lib/content/utils.js

```diff
@@ -69,6 +69,118 @@ const check = {
   },
 };
 
+/**
+ * Grab type from all content types in system
+ *
+ * @param {object} req - Express Request Object
+ *
+ * @returns {promise} reject with error message; resolve with single content-type object
+ */
+const type = (req) => {
+  return new Promise((res, rej) => {
+    const typ = req.params.type.toString();
+
+    if (!typ || typ === '') {
+      rej(config.content.messages.missing.type.replace('%type', '(no type given)'));
+    }
+
+    const exists = req.content.types.find(tp => {
+      return tp.id === typ.toLowerCase();
+    });
+
+    if (!exists || (exists === false) || (typeof exists !== 'object') || !exists.id || !exists.name) {
+      rej(config.content.messages.missing.type.replace('%type', typ));
+    }
+
+    res(exists);
+  });
+};
+
+/**
+ * Determine if an id is in uuid format
+ *
+ * @param {object} req - Express Request Object
+ *
+ * @returns {promise} reject with error message; resolve true
+ */
+const id = (req) => {
+  return new Promise((res, rej) => {
+    // no id in the request
+    if (!req.params.hasOwnProperty('id')) {
```

Comment:
if it doesn't have an ID, we're done checking this URL. 

The url is set up like this:
`/content/:type/:id/:revision/:action`
if the third section of that url is not present (`id`), then the url is going to go the `/content/:type` if there is anything after `:type` the system would default to that being checked by the `id` section as the system ignores multiple slashes. Meaning this is not a thing:
`/content/:type//:revision/:action`
it would just become this thing:
`/content/:type/:id/:revision`